lib/remap-functions/src/get_hostname.rs

```diff
@@ -0,0 +1,57 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct GetHostname;
+
+impl Function for GetHostname {
+    fn identifier(&self) -> &'static str {
+        "get_hostname"
+    }
+
+    fn compile(&self, _: ArgumentList) -> Result<Box<dyn Expression>> {
+        Ok(Box::new(GetHostnameFn))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct GetHostnameFn;
+
+impl Expression for GetHostnameFn {
+    fn execute(&self, _: &mut state::Program, _: &mut dyn Object) -> Result<Value> {
+        Ok(hostname::get()
+            .map_err(|error| format!("failed to get hostname: {}", error))?
+            .into_string()
+            .map_err(|error| format!("failed to convert hostname to string: {:?}", error))?
+            .into())
+    }
+
+    fn type_def(&self, _: &state::Compiler) -> TypeDef {
+        TypeDef {
+            kind: value::Kind::Bytes,
+            ..Default::default()
+        }
```

Comment:
Since this action can fail, we'll have to mark this as always being fallible:

```suggestion
        TypeDef {
        	fallible: true,
            kind: value::Kind::Bytes,
            ..Default::default()
        }
```